File: server/models/subject_section.go
package models

import (
	surrealModels "github.com/surrealdb/surrealdb.go/pkg/models"
)

type SubjectSectionEntity struct {
	ID            surrealModels.RecordID `json:"id" validate:"required"  swaggertype:"object"`
	ClassRoomCode *string                `json:"classroom_code"`
	SectionNumber int                    `json:"section_number"`

	SubjectOffer surrealModels.RecordID `json:"subject_offer" swaggertype:"object"`
}

File: client/src/interfaces/SubjectSection.ts
import type { Id } from './surrealDb';
import type { SubjectSchedule } from './SubjectSchedule';

export interface SubjectSection {
  id: Id<'subject_section'>;

  classroom_code: string | null;
  section_number: number;

  subject_offer: Id<'subject_offer'>;
}

export interface SubjectSectionWithSchedules extends SubjectSection {
  schedules: SubjectSchedule[];
}
